<launch>
    <arg name="pluginlists_yaml" default="$(find mavros)/launch/px4_pluginlists.yaml" />
    <arg name="config_yaml" default="$(find mavros)/launch/px4_config.yaml" />
    <group ns="uav1">
        <!--        <arg name="fcu_url" default="/dev/MasterUAV:921600"/>-->
        <arg name="fcu_url" default="/dev/NoopS0:921600"/>
        <!--        <arg name="fcu_url" default="/dev/ttyUSB0:57600"/>-->
        <arg name="gcs_url" value=""/>
        <arg name="tgt_system" value="1"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />

        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <!-- UAV2 iris_2 -->
    <group ns="usv1">
        <arg name="fcu_url" default="/dev/NoopS1:921600"/>
        <arg name="gcs_url" value=""/>
        <arg name="tgt_system" value="5"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />
        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <include file="$(find offboard)/launch/rosbag_log.launch"/>
    <node pkg = "offboard" name = "offboard" type="usv_lead_uav" output="screen"/>
</launch>